0018-connector-risk-mitigations/0018-connector-risk-mitigations.md

```diff
@@ -17,22 +17,25 @@ The main risk connectors face in Interledger is being unable to fulfill the inco
+* **Adjusting Transfer Expiry Window** - Connectors determine the window of time they require between when the incoming and outgoing transfers expire. Connectors should set this window such that they are confident that they will be able to deliver the fulfillment in time even if the outgoing transfer is executed at the last possible moment. TODO: Not sure if still relevant because we are not using HTLCs
 * **Denial of Service Protections** - Connectors should take standard [Denial of Service](https://en.bitcoin.it/wiki/Weaknesses#Denial_of_Service_.28DoS.29_attacks) attack prevention steps to prevent attackers from overloading the connectors' servers with network traffic
 * **Prioritizing Fulfillments** - Connectors should prioritize fulfilling transfers over preparing new ones or responding to quote requests and may even have separate processes or machines responsible for those different behaviors
 * **Redundant Instances** - Connectors should run redundant processes and machines to increase the difficulty of interfering with their processing
 * **Avoiding Public Endpoints** - Connectors may reduce attack vectors by using communication channels and protocols that do not entail having a publicly-accessible server (for example using [WebRTC](https://webrtc.org/) or [Virtual Private Networks](https://en.wikipedia.org/wiki/Virtual_private_network) to communicate with peers
 
 ## Liquidity Exhaustion
 
-Attackers could tie up connectors' liquidity by preparing payments through them that the attacker knows will fail. For example, an attacker could prepare numerous payments to itself and then not fulfill any of them.
+Attackers could tie up connectors' liquidity by preparing payments through them that the attacker knows will fail. For example, an attacker could prepare numerous payments to itself and then not fulfill any of them. In the case of ILPv4, this would require a significant amount of payments to cause damage because of the nature of streaming micropayments.
+
+In the worst case, during the fulfill phase a connector along the path fulfilling the payment could withhold the fulfill packet such that their corresponding connector would not receive a payment. The corresponding connector would at worst lose the unit amount being streamed in a single packet (i.e. 1 satoshi). The fulfilling connector would gain the unit amount and would put their reputation as a connector on the line and would miss out on future fee gains. Additionally, the fulfilling connector would have to close the payment channel between them and the corresponding connector, as the corresponding connector would choose not to facilitate payments through the channel after being cheated. This would result in an on-chain transaction which would make the economics for the attacker not worth it, as they could be making more by continuing to facilitate fees.
```

Comment:
I think this should be in a different section, but not sure where. Let me know what works best!

Reply:
Given the path sender -> connector A -> connector B -> receiver if connector B fails to pass along a fulfill to connector A in time, then only connector B will lose money because connector A will not update their balance with connector B (A paying B).  Therefore, a connector has nothing to gain by not passing along a fulfill because when they pass along a fulfill they are essentially getting paid back.

Reply:
Ahh, ignore this.  Didn't see thread below.